src/OrchardCore.Modules/OrchardCore.Roles/Services/RoleClaimsProvider.cs

```diff
@@ -33,14 +33,7 @@ public async Task GenerateAsync(IUser user, ClaimsIdentity claims)
             return;
         }
 
-        var isAdministrator = false;
-
-        if (await _userManager.IsInRoleAsync(user, await _systemRoleNameProvider.GetAdminRoleAsync()))
-        {
-            claims.AddClaim(StandardClaims.SiteOwner);
-
-            isAdministrator = true;
-        }
+        var isAdministrator = await _userManager.IsInRoleAsync(user, await _systemRoleNameProvider.GetAdminRoleAsync());
```

Comment:
To avoid making 2 DB calls, you can use the list of roles returned by `await _userManager.GetRolesAsync(user)` to determine whether `await _systemRoleNameProvider.GetAdminRoleAsync()` is there instead of asking the user manager/store to the check 😃